omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ConditionController2_2Test.java

```diff
@@ -0,0 +1,164 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Condition;
+import org.openmrs.ConditionClinicalStatus;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Patient;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_2;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+
+/**
+ * Tests functionality of {@link ConditionController2_2}.
+ */
+public class ConditionController2_2Test extends MainResourceControllerTest {
+	
+	private ConditionService conditionService;
+	
+	private Patient patient;
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet(RestTestConstants2_2.CONDITION_TEST_DATA_XML);
+		
+		this.conditionService = Context.getConditionService();
+		this.patient = Context.getPatientService().getPatient(2);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return "condition";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.CONDITION_UUID;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		List<Condition> activeConditions = conditionService.getActiveConditions(patient);
+		int totalConditions = 0;
+		
+		if (activeConditions != null) {
+			totalConditions = totalConditions + activeConditions.size();
+		}
+		
+		return totalConditions;
+	}
+	
+	@Test
+	public void shouldCreateACondition() throws Exception {
+		SimpleObject cond = new SimpleObject();
+		cond.add("nodeCoded", "Some condition");
+		
+		SimpleObject conditionSource = new SimpleObject();
+		conditionSource.add("condition", cond);
+		conditionSource.add("patient", patient.getUuid());
+		conditionSource.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		conditionSource.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		conditionSource.add("onsetDate", "2017-01-12 00:00:00");
+		
+		String json = new ObjectMapper().writeValueAsString(conditionSource);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newConditionSource = deserialize(handle(req));
+		Util.log("Created condition", newConditionSource);
```

Comment:
Are you using the above log for anything?

Reply:
I was using it to show that the Condition was being created correctly.